syfertext/doc.py

```diff
@@ -49,13 +49,44 @@ def set_attribute(self, name: str, value: object):
             value (object): value of the custom named attribute.
         """
 
-        # make sure there is no space in name as well prevent empty name
+        # make sure that the name is not empty and does not contains any spaces
         assert (
-            isinstance(name, str) and len(name) > 0 and (not (" " in name))
+            isinstance(name, str) and len(name) > 0 and (" " not in name)
         ), "Argument name should be a non-empty str type containing no spaces"
 
         setattr(self._, name, value)
 
+    def has_attribute(self, name: str):
+        """Checks if the Underscore object `self._` has an attribute `name`
+
+        Args:
+            name (str): name of the custom attribute.
+        
+        Returns:
+            attr_exists (bool): `True` if `self._.name` exists, otherwise `False`  
+        """
+
+        # make sure that the name is not empty and does not contains any spaces
+        assert (
+            isinstance(name, str) and len(name) > 0 and (" " not in name)
+        ), "Argument name should be a non-empty str type containing no spaces"
+
+        attr_exists = hasattr(self._, name)
+
+        return attr_exists
+
+    def remove_attribute(self, name: str):
+        """Removes the attribute `name` from the Underscore object `self._`
+
+        Args:
+            name (str): name of the custom attribute.
+        """
+
+        # Before removing it, it should exist
+        assert self.has_attribute(name), "Document does not have the attribute {}".format(name)
```

Comment:
"Before removing it, it should exist" - > "Before removing the attribute, check if it exist"
It's more readable if you dont mention things directly in 3rd person (it, they etc).
can you remove such mentions from all the comments